### PR TITLE
ML file with integrated feedback

### PR DESCRIPTION
**Pull request by TolgoAI**
Hi Max. Please have a look at an altered version of ML in accordance with your detailed feedback. Sorry for the delay.

### CODE REVIEW

**Review by maximilian-hildebrandt — APPROVED** [review]
**General:**

- Structure is mostly quite good, chapters include helpful comments on purpose of sections. Pre-processing was done mostly quite well. Good job! 2nd half of the notebook requires more work.
- The notebook you wanted to merge includes some unfinished parts (e.g., model training). Next time, do not push these to the GitHub so that we can keep a high-quality notebook in the main branch (& safe time of the reviewer ;)). 

 
**Pre-processing:**

- Overall, correct pre-processing steps were conducted. It was great that you changed the data type for columns, assessed unique values etc.
- Before removing the missing values, it would be helpful to see how many values are missing for each column. (Try out this code: df.isnull().sum(axis=0))
- If you drop missing values, add a short rationale why you decided to do that (instead of missing value imputation, for example). 
- The structure is slightly misleading: Your first step, getting familiar with the dataset, is the overall objective. Descriptive statistics, boxplots etc. are all subchapters of this first step. I would rename the first step to "The first step is to pre-process the dataset". Also, you have a 2nd pre-processing chapter before model training. You can just move the train-test-split in the model training chapter, no need for an additional chapter for this command. 
- When computing descriptive statistics, round to 2 digits to make it easier to read.

**Exploratory data analysis:**

- The data visualizations you picked are working well and help understanding the data. Try adding key results as comments to charts to extract relevant insights for the reader (e.g., comments on how the distribution looks like). What do the charts tell you? What does it indicate? What type of distribution does the data resemble (uniform, poisson, normal?). Are there any surprising patterns or data points?
- You visualize only the temperature data (by class), what about the other dimensions? Heat-preference, sex etc.? You have a section for this visualization, but it includes a generic boxplot that adds no value, since you already have a more granular boxplot above.
- The boxplot by classes requires some layout changes, currently the title and axes are hard to read.
- Correlation matrix: You cannot use a correlation matrix with the data you have, since some data is not interval-based (e.g., sex). Only use spearman correlation for interval-based data. For sex, you can compute plots of frequency to form a hypothesis of associations.
- The scatter matrix after train-test split does not work. You cannot do a scatter matrix with integer variables. I suggest to drop this element.


**Model training**

- You want to apply k-fold cross validation directly to train the model. No need for training a model without that technique.
- Do not use the test data before having completed all coding work, hyperparameter optimization etc. If you need training-test for some things, split the training data again (to have 3 subdatasets in total). 
- Select the hyperparameter values for your algorithms, which you want to optimize. Currently, you pick default values.
